app/controllers/consolidations/AssociateUCRController.scala

```diff
@@ -45,26 +46,29 @@ class AssociateUCRController @Inject()(
 
     mucrOptions match {
       case Some(mucrOptions) => Ok(associateUcrPage(associateUcr.fold(form)(form.fill), mucrOptions))
-      case None              => Redirect(controllers.routes.ChoiceController.displayPage())
+      case None              => throw ReturnToStartException
     }
   }
 
   def submit(): Action[AnyContent] = (authenticate andThen getJourney).async { implicit request =>
-    form
-      .bindFromRequest()
-      .fold(
-        formWithErrors => {
-          request.answersAs[AssociateUcrAnswers].mucrOptions match {
-            case Some(mucr) => Future.successful(BadRequest(associateUcrPage(formWithErrors, mucr)))
-            case None       => Future.successful(Redirect(controllers.routes.ChoiceController.displayPage()))
-          }
-        },
-        formData => {
-          val updatedCache = request.answersAs[AssociateUcrAnswers].copy(associateUcr = Some(formData))
-          movementRepository.upsert(Cache(request.pid, updatedCache)).map { _ =>
-            Redirect(consolidationRoutes.AssociateUCRSummaryController.displayPage())
+    val mucrOptions = request.answersAs[AssociateUcrAnswers].mucrOptions
+
+    if(mucrOptions.isEmpty) throw ReturnToStartException
```

Comment:
How about a method that extracts the mucrOptions from the request (or throws the ReturnException).  You could use it below where you need it (to re-populate the form if it fails validation) and in the display method above

Reply:
Does look better.   But there is still a "mucrOptions.get" below.

How about `processForm(mucrOptions.getOrElse(throw ReturnToStartException))`
Then processForm does not have to take an Option